framework/py/flwr/server/superlink/linkstate/in_memory_linkstate.py

```diff
@@ -330,36 +330,47 @@ def num_message_res(self) -> int:
         """
         return len(self.message_res_store)
 
-    def create_node(self, heartbeat_interval: float) -> int:
+    def create_node(self, public_key: bytes, heartbeat_interval: float) -> int:
         """Create, store in the link state, and return `node_id`."""
         # Sample a random int64 as node_id
         node_id = generate_rand_int_from_bytes(
             NODE_ID_NUM_BYTES, exclude=[SUPERLINK_NODE_ID, 0]
         )
 
         with self.lock:
-            if node_id in self.node_ids:
+            if node_id in self.nodes:
                 log(ERROR, "Unexpected node registration failure.")
                 return 0
+            if public_key in self.public_key_to_node_id:
+                raise ValueError("Public key already in use")
 
-            # Mark the node online until time.time() + heartbeat_interval
-            self.node_ids[node_id] = (
-                time.time() + heartbeat_interval,
-                heartbeat_interval,
+            # Mark the node online until now().timestamp() + heartbeat_interval
+            current = now()
+            self.nodes[node_id] = NodeInfo(
+                node_id=node_id,
+                owner_aid="",  # Unused for now
+                created_at=current.isoformat(),  # Unused for now
+                activated_at=current.isoformat(),  # Unused for now
+                deactivated_at="",  # Unused for now
```

Comment:
```suggestion
                last_activated_at=current.isoformat(),  # Unused for now
                last_deactivated_at="",  # Unused for now
```

Let's make this change? I believe https://github.com/adap/flower/pull/5968 will be merged before this PR